pkgs/top-level/all-packages.nix

```diff
@@ -14016,6 +14016,18 @@ with pkgs;
         # when adding a new linux version
         # kernelPatches.cpu-cgroup-v2."4.11"
         kernelPatches.modinst_arg_list_too_long
+
+        # https://github.com/NixOS/nixpkgs/issues/45165
+        # TODO: remove this patch once it is merged upstream and released.
+        (rec {
+          name = "hid-core-fix-grouping-by-application";
+          patch = fetchpatch {
+            name = name + ".patch";
+            # https://patchwork.kernel.org/patch/10587369/
+            url = https://git.kernel.org/pub/scm/linux/kernel/git/jikos/hid.git/patch/?id=0d6c3011409135ea84e2a231b013a22017ff999a;
```

Comment:
This patch can be fetched through either https://git.kernel.org/pub/scm/linux/kernel/git/jikos/hid.git/patch/?id=0d6c3011409135ea84e2a231b013a22017ff999a or https://patchwork.kernel.org/patch/10587369/raw/ I'm not sure which one is better to use here as I'm afraid the patchwork link might change over time, and I'm not certain this direct commit link will remain available. @NeQuissimus thoughts?

Reply:
If you're unsure, just bring the patch into the tree, so we have a local versions. I would imagine the git.kernel.org version to be the preferable solution, though.

Reply:
@NeQuissimus How about we push it as is for now? It should land in an upstream release soon anyway.

Reply:
:+1: